src/pages/index.js

```diff
@@ -74,43 +150,136 @@ previewModalCloseBtn.addEventListener("click", () => {
+let selectedCardId;
+const deleteForm = document.querySelector("#delete-form");
 
-  cardImageElement.src = data.link;
-  cardImageElement.alt = data.name;
-  cardNameElement.textContent = data.name;
+function getCardElement(cardData) {
+  const cardTemplate = document.querySelector("#card-template").content;
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
 
-  cardLikedBtn.addEventListener("click", () => {
-    cardLikedBtn.classList.toggle("card__like-btn_liked");
-  });
+  // Add a unique ID to the card
+  cardElement.id = `card-${cardData._id}`;
+
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardTitle = cardElement.querySelector(".card__title");
+  const cardLikeButton = cardElement.querySelector(".card__like-btn");
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+
+  cardImage.src = cardData.link;
+  cardImage.alt = cardData.name;
+  cardTitle.textContent = cardData.name;
 
   deleteButton.addEventListener("click", () => {
-    cardElement.remove();
+    handleDeleteCard(cardElement, cardData);
   });
 
-  cardImageElement.addEventListener("click", () => {
-    openModal(previewModal);
+  // Set initial like state
+  if (cardData.isLiked) {
+    cardLikeButton.classList.add("card__like-btn_liked");
+  }
 
-    previewModalImageElement.src = data.link;
-    previewModalImageElement.alt = data.name;
-    previewModalCaptionElement.textContent = data.name;
+  // Add click event listener for like button
+  cardLikeButton.addEventListener("click", () => {
+    handleLikeClick(cardData._id, cardLikeButton);
   });
-
   return cardElement;
 }
 
+// Open Avatar Modal
+profileAvatarBtn.addEventListener("click", () => {
+  resetValidation(avatarForm, settings);
+  openModal(avatarModal);
+});
+
+// Close Avatar Modal
+avatarModalCloseBtn.addEventListener("click", () => closeModal(avatarModal));
+
+// Avatar Form Submission
+avatarForm.addEventListener("submit", handleAvatarSubmit);
+
 function openModal(modal) {
   modal.classList.add("modal_opened");
   document.addEventListener("keydown", handleEscapeKey);
 }
 
+// Avatar
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = avatarForm.querySelector(".modal__submit-btn");
+  const avatarImg = document.querySelector(".profile__avatar");
+
+  // Set loading state
+  submitButton.disabled = true;
+  submitButton.classList.add("modal__submit-btn_loading");
+  submitButton.textContent = "Saving...";
+
+  api
+    .updateAvatar({ avatar: avatarUrlInput.value })
+    .then((userData) => {
+      avatarImg.src = "";
+      avatarImg.src = `${userData.avatar}?t=${Date.now()}`;
+
+      // Reset form and close modal
+      avatarForm.reset();
+      closeModal(avatarModal);
+
+      // Reset validation state
+      resetValidation(avatarForm, settings);
+
+      // Reset button state
+      submitButton.classList.remove("modal__submit-btn_loading");
+      submitButton.textContent = "Save";
```

Comment:
```js

      submitButton.classList.remove("modal__submit-btn_loading");
      submitButton.textContent = "Save";
```

You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.  You don't need to duplicate it in `.then` and `.catch`

**Please, correct it everywhere in the code**